api/src/test/java/bio/terra/tanagra/aousynthetic/DeviceEntityQueries.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.tanagra.aousynthetic;
+
+import static bio.terra.tanagra.aousynthetic.UnderlayUtils.ALL_DEVICE_ATTRIBUTES;
+import static bio.terra.tanagra.aousynthetic.UnderlayUtils.BQ_DATASET_SQL_REFERENCE;
+import static bio.terra.tanagra.aousynthetic.UnderlayUtils.DEVICE_ENTITY;
+import static bio.terra.tanagra.aousynthetic.UnderlayUtils.UNDERLAY_NAME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.tanagra.app.controller.EntityInstancesApiController;
+import bio.terra.tanagra.generated.model.ApiEntityDataset;
+import bio.terra.tanagra.generated.model.ApiGenerateDatasetSqlQueryRequest;
+import bio.terra.tanagra.generated.model.ApiSqlQuery;
+import bio.terra.tanagra.testing.BaseSpringUnitTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+/**
+ * Tests for device entity queries on the AoU synthetic underlay. There is no need to specify an
+ * active profile for this test, because we want to test the main application definition.
+ */
+public class DeviceEntityQueries extends BaseSpringUnitTest {
+  private static final Logger LOG = LoggerFactory.getLogger(DeviceEntityQueries.class);
+  @Autowired private EntityInstancesApiController apiController;
+
+  @Test
+  @DisplayName("correct SQL string for listing all device entity instances")
+  void generateSqlForAllDeviceEntities() {
+    ResponseEntity<ApiSqlQuery> response =
+        apiController.generateDatasetSqlQuery(
+            UNDERLAY_NAME,
+            DEVICE_ENTITY,
+            new ApiGenerateDatasetSqlQueryRequest()
+                .entityDataset(
+                    new ApiEntityDataset()
+                        .entityVariable("device_alias")
+                        .selectedAttributes(ALL_DEVICE_ATTRIBUTES)));
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+    String generatedSql = response.getBody().getQuery();
+    LOG.info(generatedSql);
+    assertEquals(
```

Comment:
opt: this PR introduces a bunch of tests that depend on exact SQL string matches. This is configured in such a way that an innocuous change in the engine code could result in a large number of tedious Java string updates across many files.

Some alternatives to consider:
1. Use a golden output files pattern here for the SQL text. Make it easy to regenerate the golden files programmatically.
2. Don't check the SQL string. Execute the queries and make targeted assertions against the responses (or use the API to do this, rather than executing it ourselves). If we're not there yet, maybe just don't bother checking the SQL string at all for now (or just look for a short substring to confirm you got the right thing).